hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -112,6 +116,10 @@ public static void createHoodieProperties(FileSystem fs, Path metadataFolder, Pr
       if (!properties.containsKey(HOODIE_ARCHIVELOG_FOLDER_PROP_NAME)) {
         properties.setProperty(HOODIE_ARCHIVELOG_FOLDER_PROP_NAME, DEFAULT_ARCHIVELOG_FOLDER);
       }
+      if (!properties.containsKey(HOODIE_TIMELINE_LAYOUT_VERSION)) {
```

Comment:
Does this mean that only new datasets will get version 1? What happens to existing datasets? They stick to version 0?

Reply:
The version in hoodie.properties is treated as default configuration. MetaClient Writers override this config with version-1 (unless they opt out). MetaClient readers will always use latest version (version - 1) which is backwards compatible. Keeping a version in hoodie.properties would also hopefully be useful in future migrations